### PR TITLE
Add Model Controller general interface

### PR DESCRIPTION
**Pull request by KharchenkoMaks**
- Made ModelController into an interface
- Moved default model controller implementation to DefaultModelController
- Moved ModelController to separate directory
- Made ModelController to be part of client UI.

### CODE REVIEW

File: src/model/controller/ModelController.h [hunk 1]
```diff
@@ -19,33 +19,25 @@
```
Comment (AntonApraksin) [inline src/model/controller/ModelController.h:14]: Remove unnecessary dependencies here

File: tests/model/DefaultModelControllerTests.cpp [hunk 2]
```diff
@@ -59,28 +59,28 @@ TEST_F(ModelControllerTests, AddTask_ShouldAskModelToAddTaskAndReturnSuccessWith
     EXPECT_CALL(*task_validator_, ValidateTask(expected_task_)).WillOnce(Return(expected_task_validator_result));
     EXPECT_CALL(*model_, AddTask(expected_task_)).WillOnce(Return(expected_model_result));
 
-    ModelController controller { std::move(model_), std::move(task_validator_), std::move(persistence_factory_) };
+    DefaultModelController controller { std::move(model_), std::move(task_validator_), std::move(persistence_factory_) };
```
Comment (AntonApraksin) [inline tests/model/DefaultModelControllerTests.cpp:62]: Consider wrapping this in `unique_ptr` and placing in `SetUp`.

**Review by AntonApraksin — CHANGES_REQUESTED** [review]

File: src/model/controller/ModelController.h (hunk 1 above)
Comment (KharchenkoMaks) [reply src/model/controller/ModelController.h:14]: moved to DefaultModelController

File: tests/model/DefaultModelControllerTests.cpp (hunk 2 above)
Comment (KharchenkoMaks) [reply tests/model/DefaultModelControllerTests.cpp:62]: will be done in refactoring PR.

**Review by AntonApraksin — APPROVED** [review]

**Review by antonovchtd — APPROVED** [review]

**Review by IlliaPlaksa — APPROVED** [review]

**Review by MichaelKupriianov — APPROVED** [review]

**Review by AntonApraksin — APPROVED** [review]

**Review by vladimir-moiseiev — APPROVED** [review]

**Pull request merged**